locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1764,24 +1769,57 @@ public void release(SshMachineLocation machine) {
         LOG.info("Releasing machine {} in {}, instance id {}", new Object[] {machine, this, instanceId});
 
         Exception tothrow = null;
+
+        if (machine instanceof JcloudsSshMachineLocation) {
+            ConfigBag setup = config().getBag();
+            for (JcloudsLocationCustomizer customizer : getCustomizers(setup)) {
+                try {
+                    customizer.preRelease((JcloudsSshMachineLocation) machine);
+                } catch (Exception e) {
+                    LOG.error("Problem invoking pre-release customizer "+customizer+" for machine "+machine+" in "+this+", instance id "+instanceId+
```

Comment:
the others were trying to release aggressively, throwing afterwards.  i've duplicated that pattern.